src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -224,8 +224,26 @@ private static void EmitCallsToMetadataProvidersForParameters(this Endpoint endp
         }
     }
 
+    public static void EmitFormAcceptsMetadata(this Endpoint endpoint, CodeWriter codeWriter)
+    {
+        if (endpoint.EmitterContext.HasFormBody)
+        {
+            var hasFormFiles = endpoint.Parameters.Any(p => p.IsFormFile);
+
+            if (hasFormFiles)
+            {
+                codeWriter.WriteLine("options.EndpointBuilder.Metadata.Add(new GeneratedAcceptsMetadata(contentTypes: GeneratedMetadataConstants.FormFileContentType));");
+            }
+            else
```

Comment:
Can both FormFile and Form be used on the same endpoint?

Reply:
They can. There is a test case for this.

Reply:
Link please

Reply:
https://github.com/dotnet/aspnetcore/blob/7f4ee4ac2fc945eab33d004581e7b633bdceb475/src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.Forms.cs#L1034

Safia added it a week or so ago. I added a few extra test cases just to make sure I was selecting the right content type based on the combination.